trie/secure_trie.go

```diff
@@ -85,6 +87,22 @@ func (t *SecureTrie) TryGetNode(path []byte) ([]byte, int, error) {
 	return t.trie.TryGetNode(path)
 }
 
+// TryUpdate account will abstract the write of an account to the
+// secure trie.
+func (t *SecureTrie) TryUpdateAccount(key []byte, acc account.Account) error {
```

Comment:
One thing that _might_ be worth taking into consideration here (?). Previously, the corresponding update method took a byteslice, which is basically 24 bytes on the stack. Now, you take an account by value,  which is `8+8+32+24` bytes (I think). 

And later on, this writes it into an rlp buffer. So essentially this means double allocations, once to copy by value, and then as before to write to rlp. 


Don't know if it makes an impact.

Reply:
yeah it should totally be a pointer, good catch